### PR TITLE
test(NODE-5043): assert MongoClients are garbage collectable

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

Backport of
https://github.com/mongodb/node-mongodb-native/pull/3553

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/tools/runner/config.ts [hunk 1]
```diff
@@ -356,7 +358,7 @@ export class TestConfiguration {
   }
 
   // Accessors and methods Client-Side Encryption
-  get mongodbClientEncryption() {
+  get mongodbClientEncryption(): typeof import('mongodb-client-encryption') {
```
Comment (durran) [inline test/tools/runner/config.ts:361]: Out of curiosity, why was this needed?

Comment (nbbeeken) [reply test/tools/runner/config.ts:361]: Not needed, I sync-ed this file with v5 which has this change. It just makes autocomplete work when you're writing an FLE test, which is nice, but unrelated to this work

**Comment by dariakp** [comment]
Marking blocked pending review of #3553

File: test/integration/node-specific/resource_tracking_script_builder.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import { expect } from 'chai';
+import { fork } from 'child_process';
+import { on, once } from 'events';
+import { readFile, unlink, writeFile } from 'fs/promises';
```
Comment (nbbeeken) [inline test/integration/node-specific/resource_tracking_script_builder.ts:4]:
```suggestion
import { promises } from 'fs';
const { readFile, unlink, writeFile } = promises;
```
You should try this.

Comment (nbbeeken) [reply test/integration/node-specific/resource_tracking_script_builder.ts]: thanks will do!

**Review by durran — APPROVED** [review]
